projects/gateway/pkg/syncer/translator_syncer.go

```diff
@@ -284,63 +310,86 @@ func (s *statusSyncer) syncStatusOnEmit(ctx context.Context) error {
 	}
 }
 
-func (s *statusSyncer) syncStatus(ctx context.Context) error {
+func (s *statusSyncer) extractCurrentReports() (reporter.ResourceReports, map[resources.InputResource]map[string]*core.Status, map[resources.InputResource]*core.Status) {
```

Comment:
nit: If we are going to have method comments (which is great and I think we should) then we should have it as a comment above the function

```suggestion
// extractCurrentReports massages several asynchronously set `statusSyncer` variables into formats consumable by `syncStatus`
func (s *statusSyncer) extractCurrentReports() (reporter.ResourceReports, map[resources.InputResource]map[string]*core.Status, map[resources.InputResource]*core.Status) {
```